### PR TITLE
AL-345: Conditional modal title

### PR DESCRIPTION
**Pull request by benoitgrasset-alma**
### Link to clickup ticket

https://app.clickup.com/t/20427503/AL-345

### Screenshots

Capture d’écran du 2022-05-04 12-34-02
Capture d’écran du 2022-05-04 12-33-43

### CODE REVIEW

File: src/Widgets/EligibilityModal/DesktopModal/index.tsx [hunk 1]
```diff
@@ -5,10 +5,10 @@ import Logo from '../components/Logo'
 import Title from '../components/Title'
 import s from './DesktopModal.module.css'
 
-const DesktopModal: FC = ({ children }) => (
+const DesktopModal: FC<{ isDeffered: boolean }> = ({ children, isDeffered }) => (
```
Comment (murdink) [inline src/Widgets/EligibilityModal/DesktopModal/index.tsx]: To fix everywhere when relevant: `deffered` -> `deferred`

File: src/Widgets/EligibilityModal/modal.test.tsx [hunk 2]
```diff
@@ -86,6 +86,23 @@ describe('Modal', () => {
...
+          onClose={() => {
+            console.log('modal closed')
+          }}
+        />,
+      )
+      expect(screen.getByTestId('modal-title-element')).toHaveTextContent(
+        'Payez en plusieurs fois ou plus tard par carte bancaire avec Alma.',
+      )
+    })
+  })
```
Comment (murdink) [inline src/Widgets/EligibilityModal/modal.test.tsx:105]: A good starting point but the test should cover both cases:
- one where the `currentPlan` value is a deferred payment plan
- one where it's not

You can use a single set of mocked data, but play with the `initialPlanIndex` to select the correct plan accordingly 

If you want to do it elegantly, you can use this type of syntax: https://jestjs.io/docs/api#2-describeeachtablename-fn-timeout

File: src/Widgets/EligibilityModal/index.tsx [hunk 3]
```diff
@@ -36,9 +36,11 @@ const EligibilityModal: FunctionComponent<Props> = ({
     status,
   }
 
+  const isDeffered = currentPlan?.deferred_days > 0 || currentPlan?.deferred_months > 0
+
   return (
     <Modal onClose={onClose} ariaHideApp={false} scrollable isOpen>
-      <ModalComponent {...modalProps}>
+      <ModalComponent {...modalProps} isDeffered={isDeffered}>
```
Comment (murdink) [inline src/Widgets/EligibilityModal/index.tsx]: I know this is out of scope, but the `{...modalProps}` code is dead, can you please remove it (+ the `modalProps` constant)

**Review by murdink — CHANGES_REQUESTED** [review]
Some comments here and there

File: src/Widgets/EligibilityModal/DesktopModal/index.tsx (hunk 1 above)
Comment (benoitgrasset-alma) [reply src/Widgets/EligibilityModal/DesktopModal/index.tsx]: done :+1:

File: src/Widgets/EligibilityModal/index.tsx (hunk 3 above)
Comment (benoitgrasset-alma) [reply src/Widgets/EligibilityModal/index.tsx]: done

File: src/Widgets/EligibilityModal/modal.test.tsx (hunk 2 above)
Comment (benoitgrasset-alma) [reply src/Widgets/EligibilityModal/modal.test.tsx:105]: done :)

Comment (murdink) [reply src/Widgets/EligibilityModal/modal.test.tsx:105]: perfect 👍

**Review by murdink — APPROVED** [review]

File: src/Widgets/EligibilityModal/modal.test.tsx [hunk 4]
```diff
@@ -57,7 +57,7 @@ describe('Modal', () => {
       expect(element).toHaveTextContent('0,00 €')
       expect(element).toHaveTextContent("Aujourd'hui")
     })
-    it('should display the the schedule for the selected payment plan', () => {
+    it('should display the schedule for the selected payment plan', () => {
```
Comment (murdink) [inline src/Widgets/EligibilityModal/modal.test.tsx:60]: thx for the typos fixes

File: src/Widgets/EligibilityModal/index.tsx [hunk 5]
```diff
@@ -28,17 +28,12 @@ const EligibilityModal: FunctionComponent<Props> = ({
...
   const currentPlan = eligiblePlans[currentPlanIndex]
-  const modalProps = {
-    eligibilityPlans: eligiblePlans,
-    currentPlanIndex,
-    setCurrentPlanIndex,
-    currentPlan,
-    status,
-  }
+
+  const isDeferred = currentPlan?.deferred_days > 0 || currentPlan?.deferred_months > 0
```
Comment (Taskim) [inline src/Widgets/EligibilityModal/index.tsx]: The spec is:

> GIVEN That at least one deferred payment fee plan is available
> WHEN Displaying the widget modale
> THEN The title should be "Payez en plusieurs fois ou plus tard par carte bancaire avec Alma." (pay later is mentioned)

So, we should not evaluate if the currentPlan is deferred, but if in all our payment plans, one of them is deferred

**Review by Taskim — CHANGES_REQUESTED** [review]

File: src/Widgets/EligibilityModal/index.tsx (hunk 5 above)
Comment (benoitgrasset-alma) [reply src/Widgets/EligibilityModal/index.tsx]: fixed :)

File: src/Widgets/EligibilityModal/index.tsx [hunk 6]
```diff
@@ -28,17 +28,14 @@ const EligibilityModal: FunctionComponent<Props> = ({
...
   const currentPlan = eligiblePlans[currentPlanIndex]
-  const modalProps = {
-    eligibilityPlans: eligiblePlans,
-    currentPlanIndex,
-    setCurrentPlanIndex,
-    currentPlan,
-    status,
-  }
+
+  const isDeferred = eligibilityPlans.some(
```
Comment (murdink) [inline src/Widgets/EligibilityModal/index.tsx]: du coup tu peux rename en `isSomePlanDeferred`

Comment (benoitgrasset-alma) [reply src/Widgets/EligibilityModal/index.tsx]: yes c'est fait

File: src/Widgets/EligibilityModal/modal.test.tsx [hunk 7]
```diff
@@ -86,7 +86,24 @@ describe('Modal', () => {
       )
     })
   })
-  describe('plans and intial index provided', () => {
+  describe('No deferred plan provided', () => {
+    it('should display the correct title, without "ou plus tard"', () => {
```
Comment (Taskim) [inline src/Widgets/EligibilityModal/modal.test.tsx]:
```suggestion
  describe('No deferred plan provided', () => {
    it('should display a different title', () => {
```

File: src/Widgets/EligibilityModal/components/Title/index.tsx [hunk 8]
```diff
@@ -5,16 +5,24 @@ import cx from 'classnames'
...
     data-testid="modal-title-element"
   >
-    <FormattedMessage
-      id="eligibility-modal.title"
-      defaultMessage="<highlighted>Payez en plusieurs fois</highlighted> ou plus tard par carte bancaire avec Alma."
-      values={{ highlighted: (...chunks: string[]) => <span>{chunks}</span> }}
-    />
+    {isSomePlanDeferred ? (
+      <FormattedMessage
+        id={'eligibility-modal.titleDeferred'}
```
Comment (Taskim) [inline src/Widgets/EligibilityModal/components/Title/index.tsx]: small detail, we use snake case everywhere for translations ids 😬

**Review by Taskim — APPROVED** [review]

**Review by gautierlabarre — APPROVED** [review]

**Pull request merged**